File: typescript/packages/bench/src/rules/revenueFloorRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class RevenueFloorRule implements RuleContract {
    readonly key = "revenue-floor";

    evaluate(data: Dataset): boolean {
        return data.revenueCents() > 0;
    }
}

File: rust/crates/atelier-bench/src/rules/revenue_floor_rule.rs
//! Rule 20/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct RevenueFloorRule;

impl RuleContract for RevenueFloorRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        data.revenue_cents() >= 0
    }
}
